packages/i18next/src/lib/functions.ts

```diff
@@ -198,17 +273,77 @@ export async function editLocalized<TKeys extends TFunctionKeys = string, TInter
 	target: Message,
 	options: LocalizedMessageOptions<TKeys, TInterpolationMap>
 ): Promise<Message>;
+/**
+ * Edits a deferred/replied interaction using the language `keys` from your i18next language setup.
+ * @since 2.4.0
+ * @param target The interaction to editReply.
+ * @param options The language keys to be sent.
+ * @example
+ * ```typescript
+ * // Using a string to specify the key to send
+ * await editLocalized(interaction, 'commands/ping:fail');
+ * // ➡ "Pong!"
+ * ```
+ */
+export async function editLocalized<TKeys extends TFunctionKeys = string>(
+	target: BaseCommandInteraction | MessageComponentInteraction,
+	keys: TKeys | TKeys[]
+): Promise<ReturnType<(BaseCommandInteraction | MessageComponentInteraction)['editReply']>>;
+/**
+ * Edits a deferred/replied interaction using an objects option.
+ * @since 2.4.0
+ * @param target The interaction to editReply.
+ * @param options The options to be sent, requiring at least `keys` to be passed.
+ * @example
+ * ```typescript
+ * // Using an object to specify the key to send
+ * await editLocalized(interaction, { keys: 'commands/ping:fail' });
+ * // ➡ "Pong!"
+ * ```
+ * @example
+ * ```typescript
+ * // Passing interpolation options into i18next
+ * const latency = 42;
+ * const took = 96;
+ *
+ * await editLocalized(interaction, {
+ * 	keys: 'commands/ping:success',
+ * 	formatOptions: { latency, took }
+ * });
+ * // ➡ "Pong! Took me 96ms to reply, and my heart took 42ms to beat!"
+ * ```
+ */
 export async function editLocalized<TKeys extends TFunctionKeys = string, TInterpolationMap extends NonNullObject = StringMap>(
-	target: Message,
-	options: TKeys | TKeys[] | LocalizedMessageOptions<TKeys, TInterpolationMap>
-): Promise<Message> {
+	target: BaseCommandInteraction | MessageComponentInteraction,
+	options: LocalizedInteractionReplyOptions<TKeys, TInterpolationMap>
+): Promise<ReturnType<(BaseCommandInteraction | MessageComponentInteraction)['editReply']>>;
+export async function editLocalized<TKeys extends TFunctionKeys = string, TInterpolationMap extends NonNullObject = StringMap>(
+	target: BaseCommandInteraction | Message | MessageComponentInteraction,
+	options: TKeys | TKeys[] | LocalizedMessageOptions<TKeys, TInterpolationMap> | LocalizedInteractionReplyOptions<TKeys, TInterpolationMap>
+): Promise<ReturnType<(BaseCommandInteraction | MessageComponentInteraction)['editReply']> | Message> {
+	// Handle Interactions:
+	if (target instanceof BaseCommandInteraction || target instanceof MessageComponentInteraction) {
+		return target.editReply(await resolveOverloads(target, options));
+	}
+
+	// Handle Message:
 	return target.edit(await resolveOverloads(target, options));
```

Comment:
I'd argue that the comments here don't really add much clarity, so they can be removed safely. After all, there are only two branches, both of which are quite self-descriptive.